avocado/core/runners/utils/messages.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import sys
+import time
+
+from ... import output
+
+
+class GenericMessage:
+    message_status = None
+
+    @classmethod
+    def _prepare_message(cls, additional_info=None):
+        """Prepare a message dict with some basic information.
+
+        This will add the keyword 'status' and 'time' to all messages.
+
+        :param: addional_info: Any additional information that you
+                               would like to add to the message.
+        :type additional_info: dict
+        :return: message dict which can be send to avocado server
+        :rtype: dict
+        """
+        status = {}
+        if isinstance(additional_info, dict):
+            status = additional_info
```

Comment:
You are saying on the docstring that `additional_info` should be a dict, and with this check here, gives us the impression that another type could be supported. Because of the docstring, Maybe you could consider using `if is not None` here.

Reply:
Yes, you are right, I will change that.